pkg/bpmn_engine/timer.go

```diff
@@ -124,10 +125,5 @@ func findExistingTimerNotYetTriggered(state *BpmnEngineState, id string, instanc
 }
 
 func findDurationValue(ice BPMN20.TIntermediateCatchEvent) (duration.Duration, error) {
-	durationStr := &ice.TimerEventDefinition.TimeDuration.XMLText
-	if durationStr == nil {
-		return duration.Duration{}, newEngineErrorf("Can't find 'timeDuration' value for INTERMEDIATE_CATCH_EVENT with id=%s", ice.Id)
```

Comment:
error message is lost

Reply:
`durationStr` could never be `nil`, so that error never actually occured.

Reply:
@GodsBoss good catch, if `durationStr` can't be `nil` - but then we need to rewrite the if clause, to capture the case when the text is empty. The goal here is to capture required and missing input as early as possible and provide good error message to the user.
If you agree, would you please bring the check back?

Reply:
@nitram509 I'd rather have this PR only change cosmetics, not behaviour. If you wish I can remove that particular change.

Reply:
My point was rather to rework this change towards a fail fast check and not removing "dead code".
Don't worry, I will merge it and then bring back the change on my own ... that round trip is shorter.